jupyterhub/apihandlers/hub.py

```diff
@@ -52,6 +52,21 @@ def post(self):
 
 
 class RootAPIHandler(APIHandler):
+    def set_default_headers(self):
+        """
+        Set any headers passed as tornado_settings['headers'].
+
+        Also responsible for setting content-type header
+        """
+        old_headers = self.settings.get("headers", {})
+        new_headers = {**old_headers}
+        new_headers.setdefault('Access-Control-Allow-Origin', '*')
+        try:
+            self.settings['headers'] = new_headers
+            super().set_default_headers()
+        finally:
+            self.settings['headers'] = old_headers
+
```

Comment:
Sorry, I was overthinking it since I thought this header had a default value, but it doesn't. Since `Access-Control-Allow-Origin` isn't set at all by default, we can call `set_header` if it's not present without any stateful patching and then move on to unmodified super:

```suggestion
        if 'Access-Control-Allow-Origin' not in self.settings.get("headers", {}):
            # allow CORS requests to this endpoint by default
            self.set_header('Access-Control-Allow-Origin', '*')
        super().set_default_headers()
```

Reply:
Ah, that's a nicer way to do this that hadn't occurred to me!